lib/ortho_functions.py

```diff
@@ -238,9 +204,381 @@
+
+                    #### Get geographic Envelope
+                    self.minlon, self.maxlon, self.minlat, self.maxlat = image_geom.GetEnvelope()
+
+                    ## Determine output image projection if applicable
+                    if type(args.epsg) is str:
+                        self.cent_lat = (self.minlat + self.maxlat) / 2
+                        self.cent_lon = (self.minlon + self.maxlon) / 2
+                        self.epsg = get_epsg_from_lat_lon(self.cent_lat, self.cent_lon, mode=args.epsg, utm_nad83=args.epsg_utm_nad83)
+                        logger.info("Automatically selected output projection EPSG code: %d", self.epsg)
+                        try:
+                            spatial_ref = utils.SpatialRef(self.epsg)
+                        except RuntimeError:
+                            logger.error(utils.capture_error_trace())
+                            logger.error("Invalid EPSG code: %i", self.epsg)
+                            rc = 1
+                        else:
+                            self.spatial_ref = spatial_ref
+
+                    #### Create target srs objects
+                    t_srs = self.spatial_ref.srs
+                    gt_ct = osr.CoordinateTransformation(g_srs, t_srs)
+
+                    #### Transform geoms to target srs
+                    if not g_srs.IsSame(t_srs):
+                        ul_geom.Transform(gt_ct)
+                        ur_geom.Transform(gt_ct)
+                        ll_geom.Transform(gt_ct)
+                        lr_geom.Transform(gt_ct)
+                        image_geom.Transform(gt_ct)
+                    logger.info("Projected extent: %s", str(image_geom))
+                    self.image_geom = image_geom
+
+                    rasterxsize_m = abs(
+                        math.sqrt((ul_geom.GetX() - ur_geom.GetX()) ** 2 + (ul_geom.GetY() - ur_geom.GetY()) ** 2))
+                    rasterysize_m = abs(
+                        math.sqrt((ul_geom.GetX() - ll_geom.GetX()) ** 2 + (ul_geom.GetY() - ll_geom.GetY()) ** 2))
+
+                    resx = abs(
+                        math.sqrt((ul_geom.GetX() - ur_geom.GetX()) ** 2 +
+                                  (ul_geom.GetY() - ur_geom.GetY()) ** 2) / xsize)
+                    resy = abs(
+                        math.sqrt((ul_geom.GetX() - ll_geom.GetX()) ** 2 +
+                                  (ul_geom.GetY() - ll_geom.GetY()) ** 2) / ysize)
+
+                    ####  Make a string for Pixel Size Specification
+                    if args.resolution is not None:
+                        self.res = "-tr {} {} ".format(args.resolution, args.resolution)
+                    else:
+                        self.res = "-tr {0:.12f} {1:.12f} ".format(resx, resy)
+                    if args.tap:
+                        self.tap = "-tap "
+
+                    logger.info("Original image size: %f x %f, res: %.12f x %.12f", rasterxsize_m, rasterysize_m, resx,
```

Comment:
Log fires on every scene. Consider changing to `logger.debug()` for opt-in verbosity.